File: web/src/api/book.ts
import http from './http';

import type { ListResponse } from '@/types/admin';

export interface BookItem {
  id: string;
  tenant_id?: string;
  title?: string;
  author?: string;
  isbn?: string;
  publisher?: string;
  publish_date?: string;
  category?: string;
  description?: string;
  status?: string;
  price?: number;
  stock_quantity?: number;
  cover_image_url?: string;
  tags?: string;
  created_at?: string;
  updated_at?: string;
}

export interface BookListQuery {
  keyword?: string;
  page?: number;
  page_size?: number;
}

const basePath = '/books';

export function listbooks(params: BookListQuery = {}): Promise<ListResponse<BookItem>> {
  return http.get<ListResponse<BookItem>>(basePath, { params });
}

export function getBook(id: string | number): Promise<BookItem> {
  return http.get<BookItem>(`${basePath}/${id}`);
}

export function createBook(data: Record<string, unknown>): Promise<BookItem> {
  return http.post<BookItem>(basePath, data);
}

export function updateBook(id: string | number, data: Record<string, unknown>): Promise<BookItem> {
  return http.put<BookItem>(`${basePath}/${id}`, data);
}

export function deleteBook(id: string | number): Promise<{ deleted: boolean }> {
  return http.delete<{ deleted: boolean }>(`${basePath}/${id}`);
}

File: server/modules/book/application/command/book.go
package command

import "time"

type CreateBook struct {
	TenantId      string    `json:"tenant_id,omitempty"`
	Title         string    `json:"title,omitempty"`
	Author        string    `json:"author,omitempty"`
	Isbn          string    `json:"isbn,omitempty"`
	Publisher     string    `json:"publisher,omitempty"`
	PublishDate   time.Time `json:"publish_date,omitempty"`
	Category      string    `json:"category,omitempty"`
	Description   string    `json:"description,omitempty"`
	Status        string    `json:"status,omitempty"`
	Price         int64     `json:"price,omitempty"`
	StockQuantity int64     `json:"stock_quantity,omitempty"`
	CoverImageUrl string    `json:"cover_image_url,omitempty"`
	Tags          string    `json:"tags,omitempty"`
}

type UpdateBook struct {
	TenantId      string    `json:"tenant_id,omitempty"`
	Title         string    `json:"title,omitempty"`
	Author        string    `json:"author,omitempty"`
	Isbn          string    `json:"isbn,omitempty"`
	Publisher     string    `json:"publisher,omitempty"`
	PublishDate   time.Time `json:"publish_date,omitempty"`
	Category      string    `json:"category,omitempty"`
	Description   string    `json:"description,omitempty"`
	Status        string    `json:"status,omitempty"`
	Price         int64     `json:"price,omitempty"`
	StockQuantity int64     `json:"stock_quantity,omitempty"`
	CoverImageUrl string    `json:"cover_image_url,omitempty"`
	Tags          string    `json:"tags,omitempty"`
}
